metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/evaluator/MetricCache.java

```diff
@@ -0,0 +1,141 @@
+package org.hypertrace.alert.engine.metric.anomaly.detector.evaluator;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.tuple.Pair;
+import org.hypertrace.alert.engine.eventcondition.config.service.v1.MetricSelection;
+import org.hypertrace.core.query.service.api.ResultSetChunk;
+import org.hypertrace.core.query.service.api.Row;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.api.ValueType;
+
+class MetricCache {
+
+  private static final int CACHE_EXPIRY_MINUTES = 5;
+
+  private final Cache<MetricSelection, MetricTimeSeries> metricCache;
+  private final QueryRequestHandler queryRequestHandler;
+
+  MetricCache(QueryRequestHandler queryRequestHandler) {
+    this.queryRequestHandler = queryRequestHandler;
+    this.metricCache =
+        CacheBuilder.newBuilder()
+            .expireAfterAccess(Duration.ofMinutes(CACHE_EXPIRY_MINUTES))
+            .build();
+  }
+
+  List<Pair<Long, Double>> getMetricValues(
+      Map<String, String> requestHeaders,
+      MetricSelection metricSelection,
+      String tenantId,
+      long startTimeMillis,
+      long endTimeMillis) {
+    long metricDurationMillis = endTimeMillis - startTimeMillis;
+
+    MetricTimeSeries metricTimeSeries = metricCache.getIfPresent(metricSelection);
+    // no record for this metric selection
+    // if requested time range is earlier than existing
+    // replace data
+    if (null == metricTimeSeries || startTimeMillis < metricTimeSeries.getStartTimeMillis()) {
+      Iterator<ResultSetChunk> iterator =
+          queryRequestHandler.executeQuery(
+              requestHeaders, metricSelection, tenantId, startTimeMillis, endTimeMillis);
+      List<Pair<Long, Double>> dataList = consumeIterator(iterator);
```

Comment:
nit (upto you) : can we rename this as `consumeIterator` -> `convertToTimeSeries` ?